### PR TITLE
[#4740] Add support for iadmin modrepl (master)

### PR DESCRIPTION
**Pull request by alanking**
Some cleanup in the mod_data_obj_meta database op.

---

CI tests running

### CODE REVIEW

File: plugins/database/src/db_plugin.cpp [hunk 1]
```diff
@@ -2141,23 +2163,26 @@ irods::error db_mod_data_obj_meta_op(
...
-            if( std::string( "resc_name") == colNames[i]) {
+    std::size_t i = 0, j = 0;
+    for (i = 0, j = 0; i < regParamNames.size(); i++) {
+        theVal = getValByKey(_reg_param, regParamNames[i].data());
+        if (theVal) {
+            if(std::string( "resc_name") == colNames[i]) {
                 continue;
             }
-            else if( std::string( "resc_hier") == colNames[i]) {
+            else if(std::string( "resc_hier") == colNames[i]) {
```
Comment (korydraughn) [inline plugins/database/src/db_plugin.cpp]: the `std::string` wrapping `"resc_hier"` is probably not needed given `colNames[i]` is a `std::string_view`.

Comment (alanking) [reply plugins/database/src/db_plugin.cpp]: Good catch - will fix

**Comment by alanking** [comment]
CI tests passed.

File: server/core/src/irodsReServer.cpp [hunk 2]
```diff
@@ -244,15 +244,7 @@ namespace {
...
-                rodsEnv env{};
-                _getRodsEnv(env);
-                auto tmp_pool = std::make_shared<irods::connection_pool>(
-                    1,
-                    env.rodsHost,
-                    env.rodsPort,
-                    env.rodsUserName,
-                    env.rodsZone,
-                    env.irodsConnectionPoolRefreshTime);
+                auto tmp_pool = irods::make_connection_pool();
```
Comment (trel) [inline server/core/src/irodsReServer.cpp]: should this be a separate commit/issue?

**Comment by trel** [comment]
looks good otherwise, # when ready.

File: server/core/src/irodsReServer.cpp (hunk 2 above)
Comment (alanking) [reply server/core/src/irodsReServer.cpp]: Yup, probably. Removed for later fixing.
